client/src/components/Inputs/Inputs.stories.js

```diff
@@ -0,0 +1,356 @@
+      </Col>
+    </Row>
+  );
+};
+
+export const dropdown = DropdownExample.bind({});
+dropdown.args = {
+  color: '',
+  label: 'Dropdown',
+  w: '100%',
+  disabled: false,
+  error: '',
+  helper: '',
+  options: [
+    { label: 'Option 1', value: 'Option 1' },
+    { label: 'Option 2', value: 'Option 2' },
+  ],
+  multi: false,
+  isOptional: false,
+};
+
+// GROUPED DROPDOWN
+const GroupedDropdownExample = (args) => {
+  const [selected, setSelected] = useState(['Option 1']);
+
+  const handleChange = (e) => {
+    setSelected(e);
+  };
+
+  return (
+    <Row>
+      <Col w={[4, 6, 4]}>
+        <Dropdown
+          {...args}
+          selected={selected}
+          handleChange={handleChange}
+          groupedOptions
+        />
+      </Col>
+    </Row>
+  );
+};
+
+export const groupDropdown = GroupedDropdownExample.bind({});
+groupDropdown.args = {
+  color: '',
+  label: 'Grouped Dropdown',
+  w: '100%',
+  disabled: false,
+  error: '',
+  helper: '',
+  options: [
+    {
+      groupLabel: 'Group 1',
+      options: [
+        { label: 'Option 1', value: 'Option 1' },
+        { label: 'Option 2', value: 'Option 2' },
+      ],
+    },
+    {
+      groupLabel: 'Group 2',
+      options: [
+        { label: 'Option A', value: 'Option A' },
+        { label: 'Option B', value: 'Option B' },
+      ],
+    },
+  ],
+  multi: false,
+  isOptional: false,
+};
+
+// MATRIX
+
+const MatrixExample = (args) => {
+  const [selected, setSelected] = useState({});
+
+  const handleChange = (selectedOption, row) => {
+    setSelected({ ...selected, [row]: selectedOption });
+  };
+
+  return (
+    <Row>
+      <Col w={[4, 6, 4]}>
+        <Matrix {...args} selected={selected} handleChange={handleChange} />
+      </Col>
+    </Row>
+  );
+};
+
+const dummyOptions = [
+  { label: 'Option 1', value: 'Option 1' },
+  { label: 'Option 2', value: 'Option 2' },
+  { label: 'Option 3', value: 'Option 3' },
+];
+
+export const matrix = MatrixExample.bind({});
+matrix.args = {
+  rows: [
+    {
+      label: 'Zoom',
+      value: 'zoom',
+      options: dummyOptions,
+    },
+    {
+      label: 'Microsoft Teams',
+      value: 'teams',
+      options: dummyOptions,
+    },
+    {
+      label: 'Google Meet',
+      value: 'meet',
+      options: dummyOptions,
+    },
+  ],
+  disabled: false,
+  errors: {},
+};
+
+// SLIDER
+
+const SliderExample = (args) => {
+  const [selected, setSelected] = useState({});
+
+  const handleChange = (e) => {
+    setSelected({ ...args.options[e], key: e });
+  };
+
+  useEffect(() => {
+    if (args.options.length % 2 === 0)
+      return setSelected({ ...args.options[0], key: 0 });
+    const middleNum = Math.floor(args.options.length / 2);
+    return setSelected({ ...args.options[middleNum], key: middleNum });
+  }, [args]);
+
+  return (
+    <Row>
+      <Col w={[4, 6, 4]}>
+        <Slider {...args} selected={selected} handleChange={handleChange} />
+      </Col>
+    </Row>
+  );
+};
+
```

Comment:
```suggestion
const SliderExample = (args) => {
  const [selected, setSelected] = useState(() => {
    if (args.options.length % 2 === 0) return { ...args.options[0], key: 0 };
    const middleNum = Math.floor(args.options.length / 2);
    return { ...args.options[middleNum], key: middleNum };
  });

  const handleChange = (e) => {
    console.log(e, '------');
    setSelected({ ...args.options[e], key: e });
  };

  return (
    <Row>
      <Col w={[4, 6, 4]}>
        <Slider {...args} selected={selected} handleChange={handleChange} />
      </Col>
    </Row>
  );
};

```

this is a vital thing for performance since `args` is a reference (object), so the useEffect will be called on each change in its parent, and this component will make unnecessary renders

Reply:
cool makes sense